.github/workflows/ci.yml

```
@@ -0,0 +1,105 @@
+---
+name: CI
+
+on:
+  push:
+  pull_request:
+  release:
+    types:
+      - published
+      - created
+      - edited
+  workflow_dispatch:
+
+jobs:
+  smoketest:
+    name: Smoke Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Fix Checkout
+        run: |
+          git fetch --force --tags
+
+      - name: Install essential
+        run: |
+          make build-dep
+
+      - name: Run tests
+        run: |
+          make lint
+
+  package_dpkg:
+    name: Package for Debian/Ubuntu
+    needs:
+      - smoketest
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Fix Checkout
+        run: |
+          git fetch --force --tags
+
+      - name: Install packages needed for build
+        run: |
+          sudo apt-get update
+          make build-dep
+
+      - name: Build
+        run: |
+          make dpkg
+
+      - name: Upload dpkg
+        uses: actions/upload-artifact@v4
+        with:
+          name: packages-dpkg
+          path: "*.deb"
+
+  upload_release:
+    name: Upload Release Assets
+    needs:
+      - package_dpkg
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Fix Checkout
+        run: |
+          git fetch --force --tags
+
+      - name: Get Tag Type
+        id: get_tagtype
+        run: |
+          TYPE=$(git cat-file -t $GITHUB_REF)
+          echo "TAGTYPE=$TYPE" >> $GITHUB_OUTPUT
+          echo ==========
+          echo REF=$GITHUB_REF
+          echo TAGTYPE=$TYPE
+
+      - name: Fetch all Artifacts
+        uses: actions/download-artifact@v4
+        with:
+          path: artifacts
+
+      - name: Upload Assets to Release
+        if: steps.get_tagtype.outputs.TAGTYPE == 'tag'
+        uses: softprops/action-gh-release@v1
+        with:
+          prerelease: true
+          files: |
+            artifacts/packages-dpkg/*.deb
+
+      - name: Upload Assets to Nexus
+        run: |
+          make deploy NEXUS_CREDS="${{ secrets.HC_NEXUS_DEPLOY }}"
```

Now that they exist, you could instead use:
```suggestion
          make deploy NEXUS_CREDS="${{ vars.NEXUS1_RW_USERNAME }}:${{ secrets.NEXUS1_RW_PASSWORD }}"
```
(For what it's worth, this would be identical to `echo "$(dcli read dl://6A3D5485-7086-44E2-8A43-7EDB7E732BD6/login):$(dcli read dl://6A3D5485-7086-44E2-8A43-7EDB7E732BD6/password)"`)

---

will fix later, in new PR